crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -164,14 +167,15 @@ impl ModuleSubscriptionActor {
         subscription: Subscribe,
     ) -> Result<(), DBError> {
         // Split logic to properly handle `Error` + `Tx`
-        let mut tx = self.relational_db.begin_tx();
+        let mut tx = self.relational_db.begin_read_tx();
 
         let result = self._add_subscription(sender, subscription, &mut tx).await;
 
         // Note: the missing QueryDebugInfo here is only used for finishing the transaction;
         // all of the relevant queries already executed, with debug info, in _add_subscription
         let ctx = ExecutionContext::subscribe(self.relational_db.address(), None);
-        self.relational_db.finish_tx(&ctx, tx, result)
+        self.relational_db.rollback_read_tx(&ctx, tx);
```

Comment:
What does it mean to rollback a read transaction?

Reply:
I guess naming could be better here, `finish_tx` looks better. It's basically to drop the guard.

Reply:
Yeah I agree, `finish` instead of `rollback` is clearer.